peppy/sample.py

```diff
@@ -371,6 +385,37 @@ def library(self):
         return self.protocol
 
 
+    def get_subsample(self, subsample_name):
+        """
+        Retrieve a single subsample by name.
+
+        :param str subsample_name: The name of the desired subsample. Should 
+            match the subsample_name column in the subannotation sheet.
+        :return SubSample: Requested SubSample object
+        """
+        subsamples = self.get_subsamples(subsample_name)
+
+        if len(subsamples) > 1:
+            _LOGGER.warn("More than one subsamples was detected; returning the first")
```

Comment:
Yeah, this is meant mostly as an experimental feature at this point... I think we haven't really formalized what should happen in the event of multiple samples...

should it throw an error, or return all samples of that name?


I guess `get_subsamples` returns all samples, but `get_subsample` throws an error, since it should always return exactly 1 sample? I don't want it sometimes returning a sample and sometimes a list, since they're different types.

Reply:
I implemented an error for duplicate types on attempting to get a subsample. you can always get both with `get_subsamples` if you need to.